timur/lib/client/jsx/contexts/query/query_where_context.tsx

```diff
@@ -0,0 +1,102 @@
+import React, {useState, createContext, useCallback} from 'react';
+
+import {QueryFilter} from './query_types';
+
+export const defaultQueryWhereParams = {
+  recordFilters: [] as QueryFilter[],
+  orRecordFilterIndices: [] as number[]
+};
+
+const defaultQueryWhereState = {
+  ...defaultQueryWhereParams
+};
+
+export type QueryWhereState = Readonly<typeof defaultQueryWhereState>;
+
+export const defaultQueryWhereContext = {
```

Comment:
Yeah, breaking this out makes some nice sense.  It is a bit more verbose to do so, but one thing I like about this approach is how easily you can go from "one big umbrella context" to get started, and then decompose into smaller systems as you build out the system and find the best places to get granular.  Context end up being fairly decomposible.

The default context bit is nice because it also gives obvious stubs for writing tests on highly integrated parts.